File: src/main/java/com/redbox/domain/notice/dto/NoticeListWrapper.java
package com.redbox.domain.notice.dto;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import lombok.Getter;

import java.util.List;

@Getter
@JsonDeserialize
public class NoticeListWrapper {
    private List<RecentNoticeResponse> notices;

    @JsonCreator
    public NoticeListWrapper() {
    }

    public NoticeListWrapper(List<RecentNoticeResponse> notices) {
        this.notices = notices;
    }
}

File: migration/src/main/kotlin/com/redbox/domain/community/notice/dto/NoticeListWrapper.kt
package com.redbox.domain.community.notice.dto

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.databind.annotation.JsonDeserialize

@JsonDeserialize
data class NoticeListWrapper(
    val notices: List<RecentNoticeResponse>
) {
    @JsonCreator
    constructor() : this(emptyList())
}
